### PR TITLE
Refactor phenotype probability comparisons

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, the model probabilities between treatment cells and control cells were compared using the ks test for each phenotype and plate. Please see the README for additional details.  The resulting comparison data and code to generate the ks comparisons are in this pr. Please let me know if you have any questions or suggestions.

### CODE REVIEW

File: 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py [hunk 1]
```diff
@@ -48,53 +48,54 @@
...
-                    "treatment_column": "gene",
-                    "Plate_Map_Name": "JUMP-Target-1_crispr_platemap"},
-               "orf":
-                   {"metadata": pd.read_csv(f"{ref_path}/JUMP-Target-1_orf_metadata.tsv", sep="\t"),
-                    "platemap": pd.read_csv(f"{ref_path}/JUMP-Target-1_orf_platemap.txt", sep="\t"),
-                    "treatment_column": "gene",
-                    "Plate_Map_Name": "JUMP-Target-1_orf_platemap"}
-                   }
+treatment_data = {
+"compound":  # Name of the treatment type
```
Comment (MikeLippincott) [inline 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py:72]: I like the inline comments for the dict. This is easy to read.

File: 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py [hunk 2]
```diff
@@ -119,30 +120,118 @@ def perform_ks_test(_dmso_probs, _treatment_probs):
...
+
+# ## Combine the model probabilty and plate data
+
+# In[7]:
+
+
+# Columns names to drop after merging data
+drop_cols = ["Assay_Plate_Barcode", "well_position"]
+
+def combine_meta(probadf):
```
Comment (MikeLippincott) [inline 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py:130]: for this function consider moving to the top of this script and adding a docstring + type hints

File: 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py [hunk 3]
```diff
@@ -119,30 +120,118 @@ def perform_ks_test(_dmso_probs, _treatment_probs):
...
+
+        # Change the treatment type data values
+        filtered_probadf["treatment_type"].replace({treatment_data[ttype]["Plate_Map_Name"]: ttype}, inplace=True)
+
+    combined_broaddf = pd.concat(broaddfs.values(), axis=0, ignore_index=True)
+
+    # Combine the probability and treatment data using the well
+    common_broaddf = pd.merge(filtered_probadf, combined_broaddf, how="inner", left_on=["Metadata_Well", "treatment_type"], right_on=["well_position", "treatment_type"])
+
+    # Drop redundant columns for the merge operations
```
Comment (MikeLippincott) [inline 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py]:
```suggestion
    # Drop redundant columns from the merge operations
```

File: 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py [hunk 4]
```diff
@@ -119,30 +120,118 @@ def perform_ks_test(_dmso_probs, _treatment_probs):
...
+treatment_data["compound"]["metadata"]["broad_sample"].fillna("DMSO", inplace=True)
+
+
+# ## Combine the model probabilty and plate data
+
+# In[7]:
+
+
+# Columns names to drop after merging data
+drop_cols = ["Assay_Plate_Barcode", "well_position"]
```
Comment (MikeLippincott) [inline 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py]: consider moving the drop cols arg into the function below

File: 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py [hunk 5]
```diff
@@ -119,30 +120,118 @@ def perform_ks_test(_dmso_probs, _treatment_probs):
...
+for proba_path in list(proba_plate_paths):
 
-treatments = sig_test.get_treatment_comparison(comp_functions, treatment_paths, probadf, barcode_platemapdf)
+    # Combine the probability data with the plate manifest files
+    common_broaddf = combine_meta(pd.read_parquet(proba_path))
+
+    # Define the phenotypes only once
+    if phenotype_cols is None:
+        # Define phenotype columns
+        phenotype_cols = common_broaddf.loc[:, "ADCCM":"SmallIrregular"].columns.tolist()
```
Comment (MikeLippincott) [inline 3.analyze_data/nbconverted/log_reg_class_balanced_areashape_analyze_well_predicted_probabilities.py:220]: Consider changing this indexing to be more dynamic. What if one of these classes are missing??? Or if another is added? outside of the range???

**Review by MikeLippincott — APPROVED** [review]
LGTM. Check the comments and consider changes. Good PR!

**Comment by MattsonCam** [comment]
Thanks Mike, merging now!

**Pull request merged**